### PR TITLE
Reporting Workflow Refactor

### LINKED ISSUE #1780: Feature: Add support for `context` in workflows
## Feature Details
Optionally Allow a user to specify a list where parameters can be found. This location should take priority over the standard hierarchy (workflow$meta, workflow$data, etc)

### PR DESCRIPTION
**Pull request by jwildfire**
## Overview

Separate reports into separate workflows and update example. See #1771  for additional context. 

Fix #1780 #1781

## Test Notes/Sample Code

Run updated example 3.2

Notes:

### CODE REVIEW

**Comment by lauramaxwell** [comment]
I'm going to hold off on fixing all the tests/checks until we have `mapping_refactor` branch merged in

File: R/RunStep.R
Comment (lauramaxwell) [inline R/RunStep.R:46]: @jwildfire Do we have an example of how this `context` field would work? it's in both `lStep` and `lData`, but it seems that the bulk of the information would be in the `lData` element. I'm just not sure i'm completely following how `context` would be used-- seeing an example may be helpful.

File: inst/workflow/metrics/cou0002.yaml
Comment (lauramaxwell) [inline inst/workflow/metrics/cou0002.yaml:15]: Should this be `Mapped_AE`, since we are doing the filtering in `Input_Rate()` now?

File: inst/workflow/data_mapping.yaml
Comment (jonthegeek) [inline inst/workflow/data_mapping.yaml:54]:
```suggestion
      strQuery: "SELECT subjid, invid, country, enrollyn FROM df"

```
Include a newline at the end of the file. Otherwise `readLines()` throws a warning. RStudio has a setting under Code > Saving, "Ensure that source files end with newline", pretty much entirely to avoid triggering that warning.

File: R/RunStep.R
Comment (jonthegeek) [inline R/RunStep.R:52]:
```suggestion
```
We don't need this unless we trigger the if conditions below.

**Review by jonthegeek — CHANGES_REQUESTED** [review]
A few changes. Still need to dig through tests.

**Comment by lauramaxwell** [comment]
@jonthegeek i'm working on getting tests to pass

File: tests/testthat/test-util-checkSpec.R
Comment (jonthegeek) [inline tests/testthat/test-util-checkSpec.R:87]: (Part of an ongoing conversation, I think)

Shouldn't this throw an error? If the columns are required and missing, isn't that an error? If not... what does "required" mean?

I think the message is wrong, too, but @lauramaxwell is probably working on that.

**Review by jonthegeek — COMMENTED** [review]
Getting close! Just a couple suggested tweaks left.

File: tests/testthat/test-util-checkSpec.R
Comment (jonthegeek) [reply tests/testthat/test-util-checkSpec.R:87]: Remaining piece of this is addressed in #1678, we can deal with it later.

Comment (lauramaxwell) [reply tests/testthat/test-util-checkSpec.R:87]: yep - fixed this in checkSpec to get the message to match correctly. not sure how that got lost

File: inst/workflow/metrics/kri0002.yaml
Comment (jwildfire) [inline inst/workflow/metrics/kri0002.yaml:30]: Let's remove strFilterNumerator and just use additional RunQuery Steps here instead

**Review by jonthegeek — APPROVED** [review]
Everything looks good to me! I think you have an unresolved comment or two to confirm, @lauramaxwell , but I'm PRETTY sure this is done!

File: inst/workflow/metrics/cou0002.yaml
Comment (lauramaxwell) [reply inst/workflow/metrics/cou0002.yaml]: yep!

**Review by lauramaxwell — APPROVED** [review]

**Pull request merged**
